pkg/validations/base.go

```diff
@@ -31,6 +34,18 @@ func RunValidations(request reconcile.Request, obj client.Object, kind string, i
 		return
 	}
 
+	// If controller has no replicas clear existing metrics and
+	// do not run any validations
+
+	replicas, err := yaml.Marshal(obj)
```

Comment:
This isn't necessary.  Would be better to use a reflect methodology similar to extract.Replicas from kube-linter.  Taking a Go object and transforming it to string to check and field isn't the right path.